File: sum_of_squares.py
"""Short program that calculates the sum of squares of a list of integers 
from a .txt file."""
import sys

def load_input(input_txt):
    """Loading input from .txt file to list"""
    with open(input_txt, 'r', encoding='utf-8') as f:
        data = [int(line.strip()) for line in f]
    return data

def sum_of_squares(input_array):
    """Returns the sum of squares of the input array."""
    return sum(x**2 for x in input_array)

if __name__ == "__main__":
    input_string = sys.argv[1]
    array = load_input(input_string)
    result = sum_of_squares(array)
    print(result)

File: sumOfSquares.c
#include <stdio.h>
#include <stdlib.h>

// Short sum of squares program that reads in a list of integers from a file 
// and calculates the sum of their squares

void sumOfSquares(int *list, int listSize,long long *result);

int main(int argc, char *argv[]) {
    
    FILE *fp = fopen(argv[1], "r");
    if (fp == NULL) { 
        printf("Error opening file.\n");
        return 1;
    }

    // Find size of input list
    int listSize = 0;
    int temp;
    while (fscanf(fp, "%d", &temp) == 1) {
        listSize++;
    }

    // Rewind filepointer and then read all numbers into an array
    rewind(fp);
    int *list = (int *)malloc(listSize * sizeof(int));
    for (int i = 0; i < listSize; i++) {
        fscanf(fp, "%d", &list[i]);
    }

    int n_loops = atoi(argv[2]);
    for (int i = 0; i < n_loops; i++) {
        // Calculate sum of squares
        long long *result = malloc(sizeof(long long));
        sumOfSquares(list, listSize, result);
        printf("Sum of squares: %lld\n", *result);
        free(result);
    }

    free(list);
    fclose(fp);
    return 0;
}

void sumOfSquares(int *list, int listSize,long long *result) {
    long long sum = 0;
    for (int i = 0; i < listSize; i++) {
        sum += (long long)list[i] * list[i];
    }
    *result = sum;
}